tests/plt/plot_test.py

```diff
@@ -0,0 +1,168 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+import scipp as sc
+from scipp.plt import Plot, Figure, widgets, Node, node
+from ..factory import make_dense_data_array, make_dense_dataset
+import matplotlib
+import ipywidgets as ipw
+
+matplotlib.use('Agg')
+
+
+def test_plot_single_1d_line():
+    da = make_dense_data_array(ndim=1)
+    n = Node(da)
+    fig = Figure(n)
+    fig.render()
+
+
+def test_plot_two_1d_lines():
+    ds = make_dense_dataset(ndim=1)
+    a = Node(ds['a'])
+    b = Node(ds['b'])
+    fig = Figure(a, b)
+    fig.render()
+
+
+def test_plot_difference_of_two_1d_lines():
+    ds = make_dense_dataset(ndim=1)
+    a = Node(ds['a'])
+    b = Node(ds['b'])
+
+    @node
+    def diff(x, y):
+        return x - y
+
+    c = diff(a, b)
+    fig = Figure(a, b, c)
+    fig.render()
+
+
+def test_plot_2d_image():
+    da = make_dense_data_array(ndim=2)
+    a = Node(da)
+    fig = Figure(a)
+    fig.render()
+
+
+def test_plot_2d_image_smoothing_slider():
+    da = make_dense_data_array(ndim=2)
+    a = Node(da)
+
+    sl = ipw.IntSlider(min=1, max=10)
+    sigma_node = widgets.WidgetNode(sl)
+
+    from scipy.ndimage import gaussian_filter
+
+    @node
+    def smooth(da, sigma):
+        out = da.copy()
+        out.values = gaussian_filter(da.values, sigma=sigma)
+        return out
+
+    smooth_node = smooth(a, sigma=sigma_node)
+    fig = Figure(smooth_node)
+    Plot([fig, sl])
+    fig.render()
+    sl.value = 5
+
+
+def test_plot_2d_image_with_masks():
+    da = make_dense_data_array(ndim=2)
+    da.masks['m1'] = da.data < sc.scalar(0.0, unit='counts')
+    da.masks['m2'] = da.coords['xx'] > sc.scalar(30., unit='m')
+
+    a = Node(da)
+
+    widget = widgets.MaskWidget(da.masks)
+    w = widgets.WidgetNode(widget)
+
+    masks_node = widgets.hide_masks(a, w)
+    fig = Figure(masks_node)
+    Plot([fig, widget])
+    fig.render()
+    widget._all_masks_button.value = False
```

Comment:
Why accessing "private" field?

Reply:
This is to simulate interaction. This should never be used in practise, and these tests will eventually disappear i suspect

Reply:
Not sure why you want to remove the tests? Don't the represent a requirement, i.e., exactly what we *should* be testing?